app/user/tabs/DataTab.tsx

```diff
@@ -0,0 +1,126 @@
+        }
+    };
+
+
+    return (
+        <>
+            <form onSubmit={handleSubmit((data) => openModal(data))} className="gap-4 space-y-12 mt-5">
+                <div className="grid grid-cols-2 gap-4">
+                    <div className="space-y-5">
+                        <FormField
+                            id="name"
+                            label='Imię'
+                            register={register}
+                            error={errors.name?.message} />
+                        <FormField
+                            id="phone_number"
+                            label="Numer telefonu"
+                            register={register}
+                            error={errors.phone_number?.message} />
+                        <FormField
+                            id="voivodship"
+                            label="Województwo"
+                            register={register}
+                            error={errors.voivodship?.message} />
+                        <FormField
+                            id="zip_code"
+                            label="Kod pocztowy"
+                            register={register}
+                            error={errors.zip_code?.message} />
+                        <FormField
+                            id="house_number"
+                            label="Numer domu"
+                            register={register}
+                            error={errors.house_number?.message} />
+                    </div>
+                    <div className="space-y-5">
+                        <FormField
+                            id="last_name"
+                            label='Nazwisko'
+                            register={register}
+                            error={errors.last_name?.message} />
+                        <FormField
+                            id="email"
+                            label='Email'
+                            register={register}
+                            error={errors.email?.message}
+                            disabled />
+                        <FormField
+                            id="city"
+                            label="Miasto"
+                            register={register}
+                            error={errors.city?.message} />
+                        <FormField
+                            id="street"
+                            label="Ulica"
+                            register={register}
+                            error={errors.street?.message} />
+                    </div>
+                </div>
+                <Button type="submit" className="px-16 py-5 font-bold">
+                    ZATWIERDŹ ZMIANY
+                </Button>
+            </form>
+
+            <ConfirmPopup
+                isOpen={isModalOpen}
+                onConfirm={handleConfirm}
+                onCancel={() => setIsModalOpen(false)} />
+            <ToastContainer />
```

Comment:
i think ToastContainer should be higher in hierarchy (maybe route layout or dashboard layout?)